### PR TITLE
feat: Remove fixed extension workarounds, align with updated JS API

### PR DESCRIPTION
**Pull request by fl0rek**

### CODE REVIEW

File: src/popup.js [hunk 1]
```diff
@@ -13,48 +13,56 @@ self.lumina = await new NodeClient(connection);
...
-	[...document.getElementsByClassName('launcher')].forEach((e) => e.disabled = true);
-
-	let started = await self.lumina.start(network_config);
-	console.log("started:", started);
-
-	updateStats();
-})
+document.getElementById("start").addEventListener("click", async (event) => {
+  let networkConfig;
+  let network = document.getElementById("network-chooser").value;
```
Comment (zvolin) [inline src/popup.js]:
```suggestion
  const networkConfig;
  const network = document.getElementById("network-chooser").value;
```

File: src/popup.js [hunk 2]
```diff
@@ -13,48 +13,56 @@ self.lumina = await new NodeClient(connection);
...
+  console.log("requesting connection to", networkConfig);
 
-async function updateStats() {
-	if (await self.lumina.is_running()) {
-		await self.lumina.wait_connected();
+  [...document.getElementsByClassName("launcher")].forEach(
+    (e) => (e.disabled = true),
+  );
+
+  let started = await self.lumina.start(networkConfig);
```
Comment (zvolin) [inline src/popup.js]:
```suggestion
  const started = await self.lumina.start(networkConfig);
```

File: src/popup.js [hunk 3]
```diff
@@ -13,48 +13,56 @@ self.lumina = await new NodeClient(connection);
...
-		document.getElementById('peers').innerText = peer_tracker_info.num_connected_peers;
-		document.getElementById('trusted-peers').innerText = peer_tracker_info.num_connected_trusted_peers;
+    document.getElementById("status").classList.remove("hidden");
 
-		let syncer_info = await self.lumina.syncer_info();
-		document.getElementById('network-head').innerText = syncer_info.subjective_head;
-		document.getElementById("stored-ranges").innerText = syncer_info.stored_headers.map((range) => {
-			return `${range.start}..${range.end}`;
-		}).join(", ");
+    let peerTrackerInfo = await self.lumina.peerTrackerInfo();
```
Comment (zvolin) [inline src/popup.js]:
```suggestion
    const peerTrackerInfo = await self.lumina.peerTrackerInfo();
```

File: src/popup.js [hunk 4]
```diff
@@ -13,48 +13,56 @@ self.lumina = await new NodeClient(connection);
...
+    document.getElementById("peers").innerText =
+      peerTrackerInfo.num_connected_peers;
+    document.getElementById("trusted-peers").innerText =
+      peerTrackerInfo.num_connected_trusted_peers;
 
-		setTimeout(updateStats, 1000);
-	} else {
-		document.getElementById('status').classList.add('hidden');
+    let syncerInfo = await self.lumina.syncerInfo();
```
Comment (zvolin) [inline src/popup.js]:
```suggestion
    const syncerInfo = await self.lumina.syncerInfo();
```

File: src/background.js [hunk 5]
```diff
@@ -1,58 +1,36 @@
...
-		// so chrome's json clone will ommit this field. Hand craft the message with expected structure as a workaround.
-		let message = { data: original_message.data };
-		sanitiseMessage(message);
-		console.debug("forwarding from worker: ", message);
-		port.postMessage(message);
-	};
+  console.log("client connected");
+  // we aren't allowed to transfer the runtime.Port we've received to worker
+  // so we patch the connection through using MessageChannel
+  let channel = new MessageChannel();
```
Comment (zvolin) [inline src/background.js]:
```suggestion
  const channel = new MessageChannel();
```

File: src/background.js [hunk 6]
```diff
@@ -1,58 +1,36 @@
...
+  port.onMessage.addListener((message) => {
+    console.debug("forwarding to worker: ", message);
+    channel.port1.postMessage(message);
+  });
+  channel.port1.onmessage = (originalMessage) => {
+    // data field isn't considered own property of MessageEvent (but is inherited from Event),
+    // so chrome's json clone will ommit this field. Hand craft the message with expected structure as a workaround.
+    let message = { data: originalMessage.data };
```
Comment (zvolin) [inline src/background.js]:
```suggestion
    const message = { data: originalMessage.data };
```

File: src/background.js [hunk 7]
```diff
@@ -1,58 +1,36 @@
-import init, { NodeClient, NodeConfig, Network } from "@fl0rek/lumina-node-wasm";
-import init_worker from './worker.js'
+import init, {
+  NodeWorker,
+  spawnNode,
+  NodeClient,
+  NodeConfig,
+  Network,
+} from "lumina-node";
```
Comment (zvolin) [inline src/background.js]:
```suggestion
import { spawnNode } from "lumina-node";
```

File: src/background.js [hunk 8]
```diff
@@ -1,58 +1,36 @@
...
 
 // must be called before using any of the functionality imported from wasm
```
Comment (zvolin) [inline src/background.js]: remove this

Comment (zvolin) [reply src/background.js]: together with `init`

File: webpack.config.js [hunk 9]
```diff
@@ -1,41 +1,38 @@
...
+      },
+      {
+        test: /\.wasm$/,
+        type: "asset/resource",
+      },
+    ],
+  },
+  experiments: {
+    asyncWebAssembly: true,
+  },
```
Comment (zvolin) [inline webpack.config.js]:
```suggestion
```

File: src/background.js (hunk 6 above)
Comment (zvolin) [inline src/background.js]: is this still an issue after we removed `sanitize message`?

File: webpack.config.js (hunk 9 above)
Comment (fl0rek) [reply webpack.config.js]: true, it seems not needed 👍

File: src/background.js (hunk 6 above)
Comment (fl0rek) [reply src/background.js]: Yes, the comment still applies.

File: src/background.js (hunk 8 above)
Comment (fl0rek) [reply src/background.js]: removed 👍

**Review by zvolin — APPROVED** [review]
nice

**Pull request merged**